src/qt/sendcoinsdialog.cpp

```diff
@@ -87,24 +88,27 @@ SendCoinsDialog::SendCoinsDialog(const PlatformStyle *_platformStyle, QWidget *p
     if (!settings.contains("bUseInstantSend"))
         settings.setValue("bUseInstantSend", false);
 
-    bool fUsePrivateSend = settings.value("bUsePrivateSend").toBool();
-    bool fUseInstantSend = settings.value("bUseInstantSend").toBool();
-    if(fLiteMode) {
+    if (!privateSendClient.fEnablePrivateSend) {
         ui->checkUsePrivateSend->setChecked(false);
         ui->checkUsePrivateSend->setVisible(false);
-        ui->checkUseInstantSend->setVisible(false);
         CoinControlDialog::coinControl->fUsePrivateSend = false;
-        CoinControlDialog::coinControl->fUseInstantSend = false;
-    }
-    else{
+    } else {
+        bool fUsePrivateSend = settings.value("bUsePrivateSend").toBool();
         ui->checkUsePrivateSend->setChecked(fUsePrivateSend);
-        ui->checkUseInstantSend->setChecked(fUseInstantSend);
         CoinControlDialog::coinControl->fUsePrivateSend = fUsePrivateSend;
-        CoinControlDialog::coinControl->fUseInstantSend = fUseInstantSend;
+        connect(ui->checkUsePrivateSend, SIGNAL(stateChanged ( int )), this, SLOT(updateDisplayUnit()));
```

Comment:
Is the move of this into the `!fEnablePrivateSend` intended? If I'm not mistaken, this was present run in all cases before.

Reply:
Same with `updateInstantSend` actually, but I assume it's ok in that case as you're moving it into the `!fLiteMode` case.

Reply:
Yes, there is no reason to use `connect` for an invisible PS checkbox (which is the case when PS is disabled, line 93).